src/lib/storage/value_segment.cpp

```diff
@@ -172,6 +172,157 @@ size_t ValueSegment<T>::estimate_memory_usage() const {
   return sizeof(*this) + _values.size() * sizeof(T) + (_null_values ? _null_values->size() * sizeof(bool) : 0u);
 }
 
+template <typename T>
+ChunkOffset ValueSegment<T>::get_non_null_begin_offset(const std::shared_ptr<const PosList>& position_filter) const {
+  Assert(_sort_order, "The segment needs to be sorted to calculate the first bound.");
+
+  ChunkOffset non_null_begin = 0;
+  if (_sort_order.value() == OrderByMode::Ascending || _sort_order.value() == OrderByMode::Descending) {
+    if (_null_values.has_value()) {
+      if (position_filter) {
+        non_null_begin = static_cast<ChunkOffset>(std::distance(
+            position_filter->cbegin(),
+            std::lower_bound(position_filter->cbegin(), position_filter->cend(), false,
+                             [&](const auto& row_id, const auto& search_value) { return row_id.is_null(); })));
+      } else {
+        non_null_begin = static_cast<ChunkOffset>(std::distance(
+            _null_values.value().cbegin(),
+            std::lower_bound(_null_values.value().cbegin(), _null_values.value().cend(), false, std::greater<bool>())));
```

Comment:
[modernize-use-transparent-functors](https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-transparent-functors.html)
```suggestion
            std::lower_bound(_null_values.value().cbegin(), _null_values.value().cend(), false, std::greater<>())));
```